src/app/views/form/form.component.ts

```diff
@@ -59,13 +65,47 @@ export class FormComponent implements OnInit {
         this.evaluationId = this.evaluation.id;
       });
     }
+    this.charsRemaining$ = concat(of(''), this.comment.valueChanges).pipe(
+      map((content) => {
+        return this.maxCharacters - (content?.length || 0);
+      })
+    );
+  }
+
+  get studentId() {
```

Comment:
They are, aren't they?

Reply:
My bad, ngOnInit is so big that I confused it with multiple functions. Another tip would be to split ngOnInit into multiple functions so it would be more clear which parts do what. For example it can be split into something like `getQueryParams`, `createForm` and some other functions.